inventree_kicad/KiCadLibraryPlugin.py

```diff
@@ -262,34 +269,44 @@ def import_meta_data(self, request):  # noqa
 
                 for field in fields:
                     if str(field.attrib.get('name', '')).lower().startswith(inventree_id_identifier.lower()):
-                        inventree_id = field.text
+                        inventree_part_id = field.text
                         break
 
                 # Missing inventree_id, cannot continue
-                if not inventree_id:
-                    logger.debug('Missing inventree_id, skipping')
-                    continue
-
-                try:
-                    inventree_id = int(inventree_id)
-                except ValueError:
-                    logger.debug('InvenTree ID is not an integer, skipping')
+                if not inventree_part_id:
+                    logger.debug('Missing part id, skipping')
                     continue
 
                 # Already checked this one
-                if inventree_id in inventree_parts:
+                if inventree_part_id in inventree_parts:
                     continue
 
                 # add to our cache, we use this to not add the same data multiple times
-                inventree_parts.add(inventree_id)
+                inventree_parts.add(inventree_part_id)
 
+                # try load part from database
+                part = None
                 try:
-                    part = Part.objects.get(id=inventree_id)
+                    part = Part.objects.get(id=inventree_part_id)
+
                 except Part.DoesNotExist:
-                    logger.debug(f'Part ID: {inventree_id} does not belong to an existing part, skipping')
-                    continue
+                    invalid_part = True
+
+                    # try also the part name if user wants it
+                    if self.get_setting('IMPORT_INVENTREE_ID_FALLBACK', None):
+                        try:
+                            part = Part.objects.get(name=inventree_part_id)
```

Comment:
shouldn't inventry_part_id be updated here since it may be used for `PartAttachment` below?

Reply:
or use `part=part` below.

Reply:
Fair point, thanks

Reply:
fixed!